BinaryEncoding.md

```diff
@@ -1,122 +1,400 @@
+| `get_local` | `0x0e` | local_index = `varuint32` | read a local variable or parameter |
+| `set_local` | `0x0f` | local_index = `varuint32` | write a local variable or parameter |
+| `load_global` | `0x10` | index = `varuint32` | * nonstandard internal opcode |
+| `store_global` | `0x11` | index = `varuint32` | * nonstandard internal opcode |
+| `call_function` | `0x12` | function_index = `varuint32` | call a function by its index |
+| `call_indirect` | `0x13` | signature_index = `varuint32` | call a function indirect with an expected signature |
+
+## Memory-related operators ([described here](AstSemantics.md#linear-memory-accesses))
+
+| Name | Opcode | Immediate | Description |
+| ---- | ---- | ---- | ---- |
+| `i32.load8_s` | `0x20` | `memory_immediate` | load from memory |
+| `i32.load8_u` | `0x21` | `memory_immediate` | load from memory  |
+| `i32.load16_s` | `0x22` | `memory_immediate` | load from memory |
+| `i32.load16_u` | `0x23` | `memory_immediate` | load from memory |
+| `i64.load8_s` | `0x24` | `memory_immediate` | load from memory |
+| `i64.load8_u` | `0x25` | `memory_immediate` | load from memory |
+| `i64.load16_s` | `0x26` | `memory_immediate` | load from memory |
+| `i64.load16_u` | `0x27` | `memory_immediate` | load from memory |
+| `i64.load32_s` | `0x28` | `memory_immediate` | load from memory |
+| `i64.load32_u` | `0x29` | `memory_immediate` | load from memory |
+| `i32.load` | `0x2a` | `memory_immediate` | load from memory |
+| `i64.load` | `0x2b` | `memory_immediate` | load from memory |
+| `f32.load` | `0x2c` | `memory_immediate` | load from memory |
+| `f64.load` | `0x2d` | `memory_immediate` | load from memory |
+| `i32.store8` | `0x2e` | `memory_immediate` | store to memory |
+| `i32.store16` | `0x2f` | `memory_immediate` | store to memory |
+| `i64.store8` | `0x30` | `memory_immediate` | store to memory |
+| `i64.store16` | `0x31` | `memory_immediate` | store to memory |
+| `i64.store32` | `0x32` | `memory_immediate` | store to memory |
+| `i32.store` | `0x33` | `memory_immediate` | store to memory |
+| `i64.store` | `0x34` | `memory_immediate` | store to memory |
+| `f32.store` | `0x35` | `memory_immediate` | store to memory |
+| `f64.store` | `0x36` | `memory_immediate` | store to memory |
+| `memory_size` | `0x3b` |  | query the size of memory |
+| `grow_memory` | `0x39` |  | grow the size of memory |
+
+The `memory_immediate` type is encoded as follows:
+
+| Name | Type | Present? | Description |
+| ---- | ---- | ---- | ---- |
+| flags | `uint8` | always | a bitfield where<br>bit `4` indicates an offset follows<br>bit `7` indicates natural alignment |
+| offset | `varuint32` | `flags[4] == 1` | the value of the offset |
+
+## Simple operators ([described here](AstSemantics#32-bit-integer-operators))
+
+| Name | Opcode | Immediate | Description |
+| ---- | ---- | ---- | ---- |
+| `i32.add` | `0x40` | | |
+| `i32.sub` | `0x41` | | |
```

Comment:
Should at least define the number of arguments for each here so the child nodes can be read.